sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/client/CachedSchemaRegistryClient.java

```diff
@@ -118,13 +122,26 @@ public SchemaRegistryObject register(
             "Registering schema. Group: '{}', name: '{}', serialization type: '{}', payload: '{}'",
                 schemaGroup, schemaName, schemaType, schemaString);
 
-        SchemaId schemaId;
+        CreateSchemaResponse response;
         try {
-            schemaId = this.restService.createSchema(schemaGroup, schemaName, schemaType, schemaString);
+            response = this.restService.createSchemaWithResponseAsync(schemaGroup, schemaName, schemaType, schemaString)
+                .block(HTTP_REQUEST_TIMEOUT);
```

Comment:
If we want to include a timeout, it must be configurable by the user?

Reply:
#12341